controllers/machine_controller_phases.go

```diff
@@ -115,6 +116,23 @@ func (r *MachineReconciler) reconcileExternal(ctx context.Context, cluster *clus
 		return external.ReconcileOutput{}, err
 	}
 
+	// With the migration from v1alpha2 to v1alpha3, Machine controllers should be the owner for the
+	// infra Machines, hence remove any existing machineset controller owner reference
+	if controller := metav1.GetControllerOf(obj); controller != nil && controller.Kind == "MachineSet" {
+		gv, err := schema.ParseGroupVersion(controller.APIVersion)
+		if err != nil {
+			return external.ReconcileOutput{}, err
+		}
+		if gv.Group == clusterv1.GroupVersion.Group {
+			ownerRefs := util.RemoveOwnerRef(obj.GetOwnerReferences(), metav1.OwnerReference{
+				APIVersion: clusterv1.GroupVersion.String(),
```

Comment:
Crit. If previous MachineSet owner has v1alpha2 version, does this not fail to remove it?
Can we add this scenario to the tests if it is not there already?

Reply:
We are just checking the Group portion of the APIVersion. I can add a separate test for the negative scenario here.

Reply:
That's a good point, can we just pass in the `controller` variable in here instead of creating a new OwnerReference? Given that we're checking the Kind and Group now, it should be safe.